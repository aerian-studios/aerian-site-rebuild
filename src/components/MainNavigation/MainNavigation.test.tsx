/// <reference types="@types/jest" />
import * as React from "react";

import * as renderer from "react-test-renderer";

import { MainNavigation } from "./index";

const pages = {allPagesJson:{
    edges: [
        {
            id: 'page1',
            path: '/page1',
            title: 'Page 1'
        },
        {
            id: 'page2',
            path: '/page2',
            title: 'Page 2'
        },
    ]
}}

describe("MainNavigation", () =>
    it("renders correctly", () => {
        const tree = renderer
            .create(<MainNavigation className="myClass" pages={pages} activePath={pages.allPagesJson.edges[1].path} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("sets '.active-path' to the matching menu item", () => {
        const tree = renderer
            .create(<MainNavigation className="myClass" pages={pages} activePath={pages.allPagesJson.edges[1].path} />)
            .toJSON();
        expect(tree).toMatchSnapshot();

        console.log(tree)
    })
);
