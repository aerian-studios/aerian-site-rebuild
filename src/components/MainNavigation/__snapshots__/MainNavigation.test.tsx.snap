// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNavigation renders correctly 1`] = `
<nav
  className=" myClass"
  id="menu-main"
>
  <span
    className="menu-item-content"
  >
    Page 1
  </span>
  <span
    className="menu-item-content"
  >
    Page 2
  </span>
</nav>
`;

exports[`MainNavigation sets '.active-path' to the matching menu item 1`] = `
<nav
  className=" myClass"
  id="menu-main"
>
  <span
    className="menu-item-content"
  >
    Page 1
  </span>
  <span
    className="menu-item-content"
  >
    Page 2
  </span>
</nav>
`;
